projects/in-progress/2024-07-11-offline-support.md

```diff
@@ -0,0 +1,48 @@
+### Problem we are solving
+
+Playground can't be used while offline.
+
+### Solution
+
+We can make Playground available offline after visiting the Playground.WordPress.net website or installing it as a PWA.
+
+### Project Goal
+
+Allow anyone to use WordPress while offline without a complicated setup.
+
+### Use Cases
+
+This will enable building offline first WordPress apps.
+
+Apps like SPAs and mobile apps that use WordPress as a backend will be able to embed the entire backend in the app and not rely on a server.
+
+### Limitations
+
+We will need to limit the switching of PHP and WordPress versions while offline to ensure users don't download too much data on the first load.
+
+## Project plan
+
+We plan to start by adding PWA support to Playground which will allow users to install it as an app and use it while online.
+
+After that, we need to ensure all assets are available after the first load which will make Playground available offline.
+
+Once Playground works offline we need to restrict some features like version switching while offline and add tests to ensure offline keeps working long-term.
+
+## Development
+
+-   [x] [PWA support](https://github.com/WordPress/wordpress-playground/pull/1086)
+-   [ ] [Backfill remote WordPress assets](https://github.com/WordPress/wordpress-playground/pull/1532)
+-   [ ] Backfill Playground assets
+    -   [ ] [Generate a list of assets to cache](https://github.com/WordPress/wordpress-playground/pull/1573)
+    -   [ ] [Cache Playground assets to enable offline support ](https://github.com/WordPress/wordpress-playground/pull/1535)
+    -   [x] [Add cache version number](https://github.com/WordPress/wordpress-playground/pull/1541)
+    -   [x] [Set web worker startup options with messages instead of query strings](https://github.com/WordPress/wordpress-playground/pull/1574)
+    -   [ ] Cache fetch requests
```

Comment:
> Let's file an issue to explore caching what we get from the plugin proxy – maybe it's a terrible idea, but maybe it would make some Blueprints work offline
from: @adamziel

Reply:
Do you mean cache API responses?

[All fetch requests to Playground.WordPress.net will be cached in the service worker](https://github.com/WordPress/wordpress-playground/pull/1535/files#diff-52dcbc798e5d0b9193b202eefd070d218be0d83560506ede1a05ef459fa40547R52). This will include plugin-proxy requests.

This makes me wonder if it's a good idea to cache API requests, or if we should be safe and cache only static files.

Reply:
I'm concerned that if we cache plugin-proxy requests the data could become outdated.
Especially because our only cache-busting mechanism is the Playground build version.

We could consider caching these requests separately from static assets and using the cache only if the user is offline.

Reply:
Perhaps this is a larger topic that also covers shipping the responses to api.w.org requests. Let’s leave it out of scope for now.

Reply:
```suggestion
    -   [ ] Cache Playground static fetch requests
```